### DISCUSSION TITLE
Upgrade from v1  to v2 documentation

### DISCUSSION THREAD

**Discussion by icfly2** [pydantic/pydantic#8312] [Question]
The upgrade documentation and bump-pedantic script are still pretty incomplete. Is there more work to come on this? There don’t appear to be any commits to those parts recently?

**Comment by samuelcolvin** [comment]
What improvements do you need to bump-pydantic?

**Reply by icfly2** [reply]
A short laundry list would be:
- actually changing .json() to .model_dump_json() and .dict() to .model_dump()
- The validator changes are nice fo rthe field validators but for validators that become a pre validator in the new one it fails with:
```
# TODO[pydantic]: We couldn't refactor the `validator`, please replace it by `field_validator` manually.
# Check https://docs.pydantic.dev/dev-v2/migration/#changes-to-validators for more information.
@validator("*", pre=True, always=False)
  ```
   which is fine, this issue is the documentation is really short on examples and details of what to do. So while I have stuff now running in most projects, I'm 100% it is not in a smart or nice way.
- I'm completely at a loss as to how to upgrade `pydantic_settings`, especially with `settings_customise_sources` .

So I don't think it has to be bump-pydantic doing the whole thing but just some more documented examples of converting a lot of the deprecated functions would be nice.
